portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,40 +26,54 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   static final String COMMENT_INPUT_ID = "user-comment";
+  static final String USERNAME_ID = "user-name";
 
   private List<String> commentsList = new ArrayList<>();
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(commentsList);
-
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
+    // Get the comment text
     String comment = getComment(request);
-
     if (comment == null) {
       return;
     }
-    commentsList.add(comment);
+    long timestamp = System.currentTimeMillis();
+    String username = getUsername(request);
+    if (username == null) {
+      return;
+    }
+
+    // Create entity and load with data
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("username", username);
+
+    // Put entity into Datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
 
     // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
   }
 
-  /** Returns the comment entered by the user. */
+  /**
+   * Returns the comment entered by the user.
+   */
   private String getComment(HttpServletRequest request) {
     // Get the input from the form.
     return request.getParameter(COMMENT_INPUT_ID);
   }
+
+  /**
+   * Returns the name entered by the user.
+   */
+  private String getUsername(HttpServletRequest request) {
+    // Get the input from the form.
+    return request.getParameter(USERNAME_ID);
+  }
```

Comment:
No need to change anything here; just something to consider for the future:
Since these functions only contain 1 line, and since that line is fairly simple, consider not making these functions and instead just call request.getParameter.

This first answer of this post does a good job of explaining when to make a one-line function: https://softwareengineering.stackexchange.com/questions/107669/one-line-functions-that-are-called-only-once

Reply:
Thank you for this explanation!